lib/memory.cpp

```diff
@@ -923,8 +932,8 @@ image_load_status_t Memory::load_elf_32(std::string image_path) {
     }
 
     // Unsupported features
-    const uint8_t ELF_ADDRESS_SIZE_32                   = 1;
-    const uint8_t ELF_TWOS_COMPLEMENT_AND_LITTLE_ENDIAN = 1;
+    [[maybe_unused]] const uint8_t ELF_ADDRESS_SIZE_32                   = 1;
+    [[maybe_unused]] const uint8_t ELF_TWOS_COMPLEMENT_AND_LITTLE_ENDIAN = 1;
     assert((file_header.e_ident[4] == ELF_ADDRESS_SIZE_32) && "Not using 32 bit address format");
```

Comment:
Wouldn't it make more sense for these to be throwing runtime error exceptions, even if we don't catch them? Or with the way everything is structured right now this wouldn't be easy to do?

Reply:
Uh well this is moreso Sam's code, I was just adding these `maybe_unused` in order to prevent errors when NDEBUG is defined.

But what you say makes sense. In fact we throw exceptions in the other loaders when the binaries are too large/write out of bounds for example. Then they are caught elsewhere and IRVE gracefully exits. We should talk to Sam about tweaking this (and then remove the `maybe_unused` once we have).